api/src/main/java/io/grpc/AdminInterface.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2021 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * Admin Interface provides a class of services for exposing the overall state of all gRPC
+ * activity in a given binary. It aims to be a convenient API that automatically determines
+ * which specific admin services are relevant.
+ */
+@ExperimentalApi("https://github.com/grpc/grpc-java/issues/7929")
+@ThreadSafe
+public final class AdminInterface {
+  private static final int DEFAULT_CHANNELZ_MAX_PAGE_SIZE = 100;
+  private static final String CHANNELZ_CLASS = "io.grpc.services.ChannelzService";
+  private static final String CSDS_CLASS = "io.grpc.xds.CsdsService";
+  private static final Logger logger = Logger.getLogger(AdminInterface.class.getName());
+  private static List<ServerServiceDefinition> standardServices;
+  private static int channelzMaxPageSize = DEFAULT_CHANNELZ_MAX_PAGE_SIZE;
+
+  // Do not instantiate.
+  private AdminInterface() {}
+
+  /**
+   * Returns a list of gRPC's built-in admin services.
+   *
+   * @return list of standard admin services
+   */
+  public static synchronized List<ServerServiceDefinition> getStandardServices() {
+    if (standardServices == null) {
+      List<ServerServiceDefinition> services = new ArrayList<>();
+      ServerServiceDefinition channelz = loadService(
+          CHANNELZ_CLASS, new Class<?>[]{int.class}, new Object[]{channelzMaxPageSize});
+      if (channelz != null) {
+        services.add(channelz);
+      }
+      ServerServiceDefinition csds = loadService(CSDS_CLASS, new Class<?>[]{}, new Object[]{});
+      if (csds != null) {
+        services.add(csds);
+      }
+      standardServices = Collections.unmodifiableList(services);
+    }
+    return standardServices;
+  }
+
+  /**
+   * Initialize the size of Channelz response pages. Calls after the first call of {@link
+   * #getStandardServices} are no-op.
+   *
+   * @param size the max page size
+   */
+  public static synchronized void setChannelzMaxPageSize(int size) {
```

Comment:
I don't remember this being part of the design. Are other languages going to have something similar, or did you decide to do this yourself?

This is something that would be hard if we swapped to an SPI approach.

Reply:
Ah, I made this up. Looks the configurable page size is Java specific. Other languages do not have such a knob.

We may just want to hardcode a number then.

Reply:
Hard-coding SGTM

Reply:
Hardcoded to 100. Or we can set to infinite to match other languages not having such knob. Doesn't matter much...